src/kernels/jupyter/launcher/serverUriStorage.ts

```diff
@@ -149,6 +150,13 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
             Settings.JupyterServerRemoteLaunchUriListKey,
             undefined
         );
+
+        // Notify out that we've removed the list to clean up controller entries, passwords, ect
+        this._onDidRemoveUris.fire(
```

Comment:
I waffled on just calling removeUri() function per each item in the list and leaving the event as is. But the remove operation felt a bit different to me (such as removeURI() watching the active URI and setting back to local on removal, which we don't want here) so I moved this instead to a bulk event here instead of firing a chain of the other single removal events.